### PR TITLE
add delete method, so that box wouldnt stack with junk mails

### PR DESCRIPTION
**Pull request by Taxanas**

### CODE REVIEW

File: src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java [hunk 1]
```diff
@@ -405,4 +405,13 @@ protected Boolean moveMessage(Message message, String toFolderPath, boolean crea
             throw new SlimFixtureException("Unable to move the message", ex);
         }
     }
+    public void markAsDeleted() {
+        if (lastMessage != null) {
+            try {
+                lastMessage.setFlag(Flags.Flag.DELETED, true);
+            } catch (MessagingException e) {
+                throw new StopTestException("failed to mark as deleted: " + e);
```
Comment (fhoeben) [inline src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java]: Other methods throw SlimFixtureException, why did you choose StopTest here?
Please use overload with accepts cause as 2nd parameter instead just appending String version of exception.

File: src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java [hunk 2]
```diff
@@ -405,4 +405,13 @@ protected Boolean moveMessage(Message message, String toFolderPath, boolean crea
             throw new SlimFixtureException("Unable to move the message", ex);
         }
     }
+    public void markAsDeleted() {
```
Comment (fhoeben) [inline src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java]: Would `markLastMessageAsDeleted` not be a better name (in line with `moveLastMessageToFolder`)?

File: src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java [hunk 3]
```diff
@@ -405,4 +405,13 @@ protected Boolean moveMessage(Message message, String toFolderPath, boolean crea
...
+        if (lastMessage != null) {
```
Comment (fhoeben) [inline src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java:413]: Should an exception be thrown if there is no last message? I.e. `throw new SlimFixtureException(false, "There is no last message to mark as deleted");`

**Review by fhoeben — CHANGES_REQUESTED** [review]
Seems like a good feature. I always expected to move messages to 'Trash' mailbox.

Some comments/questions though...

File: src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java (hunk 2 above)
Comment (Taxanas) [reply src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java]: Sure, will do ;)

File: src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java (hunk 3 above)
Comment (Taxanas) [reply src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java:413]: You are right ;)

File: src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java (hunk 1 above)
Comment (Taxanas) [reply src/main/java/nl/hsac/fitnesse/fixture/slim/EmailFixture.java]: Sorry, my bad. Thank you for your patience  :)

**Review by Taxanas — COMMENTED** [review]
Changes done

**Pull request merged**